src/compute/src/logging/differential.rs

```diff
@@ -115,34 +115,45 @@ pub(super) fn construct<A: Allocate>(
 
         // Encode the contents of each logging stream into its expected `Row` format.
         let mut packer = PermutedRowPacker::new(DifferentialLog::ArrangementBatches);
-        let arrangement_batches = batches.as_collection().map(move |op| {
-            packer.pack_slice(&[
-                Datum::UInt64(u64::cast_from(op)),
-                Datum::UInt64(u64::cast_from(worker_id)),
-            ])
-        });
+        let arrangement_batches = batches
+            .as_collection()
+            .mz_arrange_core::<_, RowSpine<_, _, _, _>>(
+                Exchange::new(move |_| u64::cast_from(worker_id)),
```

Comment:
Not that this matters much, but the `Exchange`s at this position are merely defensive, right? In the previous version of the code, the data were just pipelined, so using a `Pipeline` pact here would also be an option.

Reply:
Yep, and thinking about it I'm not sure it makes a lot of sense! But I don't want to change it as part of this PR because there are other places that use the same construct, and it's better to change all at the same time.

The exchange would go away if/once #22384 lands because then we make it an invariant that log data gets processed locally.

Reply:
FWIW, this is where we decided to keep these exchanges: https://github.com/MaterializeInc/materialize/pull/18062#discussion_r1134382604.

But I also find `Pipeline` more readable here, as it conveys what invariants we believe to be true. When seeing the exchange without an explanation you might start to doubt your understanding of how events flow through these dataflows.